src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -888,6 +888,12 @@ void G1CollectedHeap::resize_heap(size_t resize_bytes, bool should_expand) {
 
 void G1CollectedHeap::resize_heap_after_full_collection(size_t allocation_word_size) {
   assert_at_safepoint_on_vm_thread();
+  // Short-cut calculation if the heap size is fixed (MinHeapSize == MaxHeapSize) and
+  // cannot expand or shrink.
+  if (is_fixed_size_heap()) {
+    log_debug(gc, ergo, heap)("Skip heap resize after full collection.");
```

Comment:
The comment should explain why we skip too. I.e. `Skip heap resize after full collection because heap size is fixed` or so.